crates/floresta-chain/src/pruned_utreexo/consensus.rs

```diff
@@ -233,14 +380,51 @@ impl Consensus {
         };
         Ok(())
     }
-    #[allow(unused)]
+    /// Validate Locktime. if mtp is 0, skip the validation.
     fn validate_locktime(
         input: &TxIn,
         transaction: &Transaction,
+        out_map: &UtxoMap,
         height: u32,
+        mtp: u32,
     ) -> Result<(), BlockValidationErrors> {
-        unimplemented!("validate_locktime")
+        if mtp == 0 {
+            return Ok(());
+        }
+        if input.sequence.is_relative_lock_time() {
+            //validate lock time
+            let prevout = out_map.get(&input.previous_output).unwrap();
+            if input.sequence.is_height_locked() {
+                let commited_height = prevout.get_height();
+                //to retrieve the span contained in the sequence we have to just get the u32 value that the sequence contains.
+                let height_span = input.sequence.0;
+
+                // if the commited height + the span is greater than the current height, the transaction is invalid.
+                if commited_height + height_span > height {
+                    return Err(BlockValidationErrors::BadRelativeLockTime);
+                }
+            }
+            if input.sequence.is_time_locked() {
+                let commited_time = prevout.get_time();
+                // here we have to shift the sequence 16 bits to the left and 16 to the right get the time lock without the flag messing with us.
+                let time_lock = ((input.sequence.0 << 16) >> 16) * 512 as u32;
+
+                if commited_time + time_lock > mtp {
+                    return Err(BlockValidationErrors::BadRelativeLockTime);
+                }
+            }
+        }
+        if input.sequence.enables_absolute_lock_time() {
+            if !transaction.is_absolute_timelock_satisfied(
```

Comment:
Why using rust-bitcoin's method here and not above?

Reply:
overhead. Extracting the value directly from the sequence instead of casting bureaucratic types in rust-bitcoin is simpler. But surely is a negligible difference in performance, etc.

Also there is this [Issue](https://github.com/rust-bitcoin/rust-bitcoin/issues/3334) on rust-bitcoin that i think that can happen an api-break in this function. Surely the locktime checks we are implementing will be reviewed solving https://github.com/vinteumorg/Floresta/issues/197